chalk-solve/src/rust_ir.rs

```diff
@@ -168,6 +168,15 @@ pub struct FnDefDatumBound<I: Interner> {
     pub where_clauses: Vec<QuantifiedWhereClause<I>>,
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct ClosureDatum<I: Interner> {
+    pub id: ClosureId<I>,
+
+    pub inputs_and_output: Binders<FnDefInputsAndOutputDatum<I>>,
```

Comment:
probably worth pointing people at `inputs_and_output` from `FnDef` to explain the late-bound business